master/internal/trial.go

```diff
@@ -382,6 +383,13 @@ func (t *trial) runningReceive(ctx *actor.Context) error {
 
 	case actor.ChildStopped:
 
+	case *apiv1.KillTrialRequest:
+		t.killed = true
+		if t.task != nil {
+			ctx.Tell(t.rp, scheduler.TerminateTask{TaskID: t.task.ID, Forcible: true})
+		}
```

Comment:
blocking: I know it's like 4 lines, but since this is repeated from `case killTrial`, can we pull it into a method and call it.

Taking it a little further, tbh, I'm not a _huge_ fan of sending the request objects directly to the actors themselves; i think it'd be cleaner to convert from (request model) `KillTrialRequest` -> (internal model) `killTrial` in the controller and just have the trial actor know how to handle a `killTrial` message. It doesn't really need to be aware of the proto objects. But I know this pattern is already taking hold a bit. Thoughts?

Reply:
I swtiched back and forth on that. I think the idea was to discourage changes to the old enpoint when we are changing the new endpoint but it doesn't make that much sense now :thinking: . I'm reusing it now

Reply:
about defining and setting message types do you think we could go the other way? as in have the agent communications use the same proto definitions? I don't remember if we were planning to use GRPC for inter-agent communication as well or not. in either case I think this is a good topic for sync.